aiohttp/cookiejar.py

```diff
@@ -261,8 +261,43 @@ def filter_cookies(self, request_url: URL = URL()) -> "BaseCookie[str]":
                 request_origin = request_url.origin()
             is_not_secure = request_origin not in self._treat_as_secure_origin
 
+        # d: domain, d could be subdomain
+        # p: path
+        d = hostname
+
+        pairs = set()
+        while d:
+            p = request_url.path
+            while p:
+                pairs.add((d, p))
+                path_leftovers = p.rsplit("/", maxsplit=1)
+                # handle last element for rsplit
+                if len(path_leftovers) > 1:
+                    p = path_leftovers[0]
+                else:
+                    p = None
```

Comment:
Not sure I understand why this is needed.

Isn't a path always starting with `/`, so we'll get to `/foo` which will be split to `["", "foo"]`. `""` will evaluate as False, so the while loop will break (before it ever reaches this else condition).